### PR TITLE
Design community show page

### PR DESCRIPTION
**Pull request by vasanthakumar-a**
1. Search button in navbar
2. join button in community page
3. Leave Community feature
4. Subscription show page feature
5. Designed community home page
6. Create Post as input field in community page

### CODE REVIEW

File: app/views/communities/index.js.erb [hunk 1]
```diff
@@ -4,4 +4,14 @@ $("<%=j render partial: "list"%>").appendTo($("#my-communities"));
...
+
+$(window).scroll(function() {
+  var next_url = $("#infinite-scrolling a[rel='next']").attr('href');
+  if (( next_url ) && ((window.innerHeight + window.pageYOffset) >= document.body.offsetHeight))
+  {
+    $('#my-communities').show();
+    $.getScript(next_url);
+    return;
+  }
+});
```
Comment (iMagesh) [inline app/views/communities/index.js.erb]: is this index.js.erb file really necessary?
can we move this to application.js as a file so it gets loaded the first time when the site loads?

File: app/views/communities/show.html.erb [hunk 2]
```diff
@@ -2,54 +2,85 @@
   <%= image_tag(@community.cover_image, height: "200px", width: "1200px") %>
 <% else %>
   <%= link_to "#" do %>
-    <img height= "200px", width= "1200px" src="https://cdn.cbeditz.com/cbeditz/preview/solid-dark-blue-gradient-background-wallpapers-70-116145018144hkixikfuh.jpg">
+    <img height= "170px", width= "100%" src="https://cdn.cbeditz.com/cbeditz/preview/solid-dark-blue-gradient-background-wallpapers-70-116145018144hkixikfuh.jpg">
```
Comment (iMagesh) [inline app/views/communities/show.html.erb:5]: if this is the cover image, instead of creating this using img tag can you add it through css?

Comment (iMagesh) [reply app/views/communities/show.html.erb:5]: also don't use external url for assets, instead download the image and load it locally

File: app/views/communities/show.html.erb [hunk 3]
```diff
@@ -2,54 +2,85 @@
...
-      <%= link_to "#" do %>
-        <img class="profile-pic" src="https://www.iconpacks.net/icons/2/free-reddit-logo-icon-2436-thumb.png">
+<div class="container">
+  <div class="row">
+    <div class="col-sm-1 mr-3">
+      <% if @community.profile_image.attached? %>
+        <%= image_tag(@community.profile_image, class: "profile-pic") %>
+      <% else %>
+        <%= link_to "#" do %>
+          <img class="profile-pic" src="https://www.iconpacks.net/icons/2/free-reddit-logo-icon-2436-thumb.png" class="profile-pic">
```
Comment (iMagesh) [inline app/views/communities/show.html.erb:15]: don't use external url for assets, instead download the image and load it locally

File: app/views/public/index.js.erb [hunk 4]
```diff
@@ -4,4 +4,14 @@ $("<%=j render partial: "posts/list", locals: { posts:@posts } %>").appendTo($("
...
+
+$(window).scroll(function() {
+  var next_url = $("#infinite-scrolling a[rel='next']").attr('href');
+  if (( next_url ) && ((window.innerHeight + window.pageYOffset) >= document.body.offsetHeight))
+  {
+    $('#my-posts').show();
+    $.getScript(next_url);
+    return;
+  }
+});
```
Comment (iMagesh) [inline app/views/public/index.js.erb:17]: this seems to be repeated in many places. Can we just move this to one file and use it everywhere?

**Review by iMagesh — CHANGES_REQUESTED** [review]
Please read and respond to all the comments. Make the necessary changes and notify me.

File: app/views/communities/index.js.erb (hunk 1 above)
Comment (tusharravindran) [reply app/views/communities/index.js.erb]: we have created infinite scrolling function to application.js

**Comment by tusharravindran** [comment]
all the  Conflicts have been Resolved. Submision for Merge Request

File: app/views/layouts/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    for (i = 0; i < a.length; i++) {
+      txtValue = a[i].textContent || a[i].innerText;
+      if (txtValue.toUpperCase().indexOf(filter) > -1) {
+        a[i].style.display = "";
+      } else {
+        a[i].style.display = "none";
+      }
+    }
+  }
+</script>
```
Comment (iMagesh) [inline app/views/layouts/_navbar.html.erb:89]: can you move this to a separate js file?

**Review by iMagesh — CHANGES_REQUESTED** [review]
i have added one comment please fix that and let me know. We can merge.

File: app/views/layouts/_navbar.html.erb (hunk 5 above)
Comment (vasanthakumar-a) [reply app/views/layouts/_navbar.html.erb]: Resolved bug - Navbar js is moved to separate file

**Comment by vasanthakumar-a** [comment]
The Conflict have been Resolved.

**Pull request merged**